.delivery/build-cookbook/recipes/functional.rb

```diff
@@ -31,3 +31,38 @@
     end
   end
 end
+
+# Teardown Acceptance, Union, and Reheasal Omnitruck instances.
+if node['delivery']['change']['stage'] == 'delivered'
```

Comment:
you should use `workflow_stage?('delivered')` here rather than hitting the node attribute directly.